src/mlpack/methods/kernel_svm/kernel_svm.hpp

```diff
@@ -0,0 +1,213 @@
+/**
+ * @file methods/kernel_svm/kernel_svm.hpp
+ * @author Himanshu Pathak
+ *
+ * An implementation of Kernel SVM.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_KERNEL_SVM_KERNEL_SVM_HPP
+#define MLPACK_METHODS_KERNEL_SVM_KERNEL_SVM_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <ensmallen.hpp>
+#include <string>
+
+
+namespace mlpack {
+namespace svm {
+
+/**
+ * The KernelSVM class.
+ *
+ *
+ * @tparam MatType Type of data matrix.
+ */
+template <typename MatType = arma::mat, typename KernelType = kernel::LinearKernel>
+class KernelSVM
+{
+ public:
+  /**
+   * Construct the Kernel SVM class with the provided data and labels.
+   *
+   * @tparam OptimizerType Desired differentiable separable optimizer
+   * @param data Input training features. Each column associate with one sample
+   * @param labels Labels associated with the feature data.
+   * @param numClasses Number of classes for classification.
+   * @param delta Margin of difference between correct class and other classes.
+   * @param fitIntercept add intercept term or not.
+   * @param optimizer Desired optimizer.
+   */
+  KernelSVM(const MatType& data,
+            const arma::Row<size_t>& labels,
+            const size_t numClasses = 2,
+            const double delta = 1.0,
+            const double C = 1.0,
+            const std::string& kernelFunction = "linear",
```

Comment:
Can we use `const KernelType& kernel = KernelType()` here instead? That way we can reuse the existing kernel implementation, I think @rcurtin suggested this at some point.